--- libcxx/src/support/runtime/exception_libcxxabi.ipp.bak	2020-08-08 15:33:50.305572263 +0000
+++ libcxx/src/support/runtime/exception_libcxxabi.ipp	2020-08-08 15:34:00.981684599 +0000
@@ -17,11 +17,7 @@
 
 int uncaught_exceptions() _NOEXCEPT
 {
-# if _LIBCPPABI_VERSION > 1001
-    return __cxa_uncaught_exceptions();
-# else
     return __cxa_uncaught_exception() ? 1 : 0;
-# endif
 }
 
 } // namespace std
