From edba2e1ab9738a260363d47ad0e4b69f5d38ea25 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 26 May 2024 13:35:29 +1100
Subject: [PATCH 7/7] link libunwind with dl & pthread also on linux

---
 libunwind/src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libunwind/src/CMakeLists.txt b/libunwind/src/CMakeLists.txt
index 2cf2c8c0f1f7..7a8ab6312929 100644
--- a/libunwind/src/CMakeLists.txt
+++ b/libunwind/src/CMakeLists.txt
@@ -78,11 +78,11 @@ else()
   add_library_flags_if(LIBUNWIND_HAS_C_LIB c)
 endif()
 
-if (NOT APPLE)
+if (TRUE)
   add_library_flags_if(LIBUNWIND_HAS_DL_LIB dl)
 endif()
 
-if (LIBUNWIND_ENABLE_THREADS AND NOT APPLE)
+if (LIBUNWIND_ENABLE_THREADS)
     add_library_flags_if(LIBUNWIND_HAS_PTHREAD_LIB pthread)
 endif()
 
